pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -291,52 +320,51 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
-            // initialized. Thus, the Router treats it as a non-ERC4626 token.
-            if (address(underlyingToken) == address(0)) {
-                if (wrappedAmountsOut[i] < params.minAmountsOut[i]) {
-                    revert IVaultErrors.AmountOutBelowMin(
-                        erc4626PoolTokens[i],
-                        wrappedAmountsOut[i],
-                        params.minAmountsOut[i]
-                    );
+            if (useWrappedTokens[i] == true) {
+                amountsOut[i] = wrappedAmountsOut[i];
+
+                if (amountsOut[i] < params.minAmountsOut[i]) {
+                    revert IVaultErrors.AmountOutBelowMin(erc4626PoolTokens[i], amountsOut[i], params.minAmountsOut[i]);
                 }
 
-                underlyingAmountsOut[i] = wrappedAmountsOut[i];
                 if (isStaticCall == false) {
-                    _sendTokenOut(params.sender, erc4626PoolTokens[i], underlyingAmountsOut[i], params.wethIsEth);
+                    _sendTokenOut(params.sender, erc4626PoolTokens[i], amountsOut[i], params.wethIsEth);
                 }
-                continue;
-            }
+            } else {
+                require(
+                    address(underlyingToken) != address(0),
+                    "CompositeLiquidityRouter: ERC4626 buffer not initialized"
+                );
 
-            // `erc4626BufferWrapOrUnwrap` will fail if the wrappedToken is not ERC4626-conforming.
-            (, , underlyingAmountsOut[i]) = _vault.erc4626BufferWrapOrUnwrap(
-                BufferWrapOrUnwrapParams({
-                    kind: SwapKind.EXACT_IN,
-                    direction: WrappingDirection.UNWRAP,
-                    wrappedToken: wrappedToken,
-                    amountGivenRaw: wrappedAmountsOut[i],
-                    limitRaw: params.minAmountsOut[i]
-                })
-            );
+                // `erc4626BufferWrapOrUnwrap` will fail if the wrappedToken is not ERC4626-conforming.
+                (, , amountsOut[i]) = _vault.erc4626BufferWrapOrUnwrap(
+                    BufferWrapOrUnwrapParams({
+                        kind: SwapKind.EXACT_IN,
+                        direction: WrappingDirection.UNWRAP,
+                        wrappedToken: wrappedToken,
+                        amountGivenRaw: wrappedAmountsOut[i],
+                        limitRaw: params.minAmountsOut[i]
+                    })
+                );
 
-            if (isStaticCall == false) {
-                _sendTokenOut(params.sender, underlyingToken, underlyingAmountsOut[i], params.wethIsEth);
+                if (isStaticCall == false) {
+                    _sendTokenOut(params.sender, underlyingToken, amountsOut[i], params.wethIsEth);
+                }
             }
         }
     }
 
     /// @dev Assumes array lengths have been checked externally.
-    function _wrapTokens(
+    function _prepareTokens(
```

Comment:
I'd just be gigaexplicit with this function name. How about `_wrapTokensIfRequired`?